From patchwork Tue Mar 14 00:38:28 2023
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Jonas Karlman <jonas@kwiboo.se>
X-Patchwork-Id: 1756641
X-Patchwork-Delegate: ykai007@gmail.com
Return-Path: <u-boot-bounces@lists.denx.de>
X-Original-To: incoming@patchwork.ozlabs.org
Delivered-To: patchwork-incoming@legolas.ozlabs.org
Authentication-Results: legolas.ozlabs.org;
 spf=pass (sender SPF authorized) smtp.mailfrom=lists.denx.de
 (client-ip=85.214.62.61; helo=phobos.denx.de;
 envelope-from=u-boot-bounces@lists.denx.de; receiver=<UNKNOWN>)
Authentication-Results: legolas.ozlabs.org;
	dkim=pass (2048-bit key;
 unprotected) header.d=kwiboo.se header.i=@kwiboo.se header.a=rsa-sha256
 header.s=s1 header.b=ipoZi0uj;
	dkim-atps=neutral
Received: from phobos.denx.de (phobos.denx.de [85.214.62.61])
	(using TLSv1.3 with cipher TLS_AES_256_GCM_SHA384 (256/256 bits)
	 key-exchange X25519 server-signature ECDSA (P-384))
	(No client certificate requested)
	by legolas.ozlabs.org (Postfix) with ESMTPS id 4PbF6t5NM8z247D
	for <incoming@patchwork.ozlabs.org>; Tue, 14 Mar 2023 11:40:22 +1100 (AEDT)
Received: from h2850616.stratoserver.net (localhost [IPv6:::1])
	by phobos.denx.de (Postfix) with ESMTP id E95CD860F3;
	Tue, 14 Mar 2023 01:39:21 +0100 (CET)
Authentication-Results: phobos.denx.de;
 dmarc=pass (p=reject dis=none) header.from=kwiboo.se
Authentication-Results: phobos.denx.de;
 spf=pass smtp.mailfrom=u-boot-bounces@lists.denx.de
Authentication-Results: phobos.denx.de;
	dkim=pass (2048-bit key;
 unprotected) header.d=kwiboo.se header.i=@kwiboo.se header.b="ipoZi0uj";
	dkim-atps=neutral
Received: by phobos.denx.de (Postfix, from userid 109)
 id 9ECC686145; Tue, 14 Mar 2023 01:38:40 +0100 (CET)
X-Spam-Checker-Version: SpamAssassin 3.4.2 (2018-09-13) on phobos.denx.de
X-Spam-Level: 
X-Spam-Status: No, score=-0.8 required=5.0 tests=BAYES_00,DKIM_SIGNED,
 DKIM_VALID,DKIM_VALID_AU,DKIM_VALID_EF,RCVD_IN_BL_SPAMCOP_NET,
 RCVD_IN_MSPIKE_H2,SPF_HELO_NONE,SPF_PASS,UNPARSEABLE_RELAY
 autolearn=no autolearn_force=no version=3.4.2
Received: from wrqvtzvf.outbound-mail.sendgrid.net
 (wrqvtzvf.outbound-mail.sendgrid.net [149.72.126.143])
 (using TLSv1.3 with cipher TLS_AES_128_GCM_SHA256 (128/128 bits))
 (No client certificate requested)
 by phobos.denx.de (Postfix) with ESMTPS id 7EBB585E71
 for <u-boot@lists.denx.de>; Tue, 14 Mar 2023 01:38:30 +0100 (CET)
Authentication-Results: phobos.denx.de;
 dmarc=pass (p=reject dis=none) header.from=kwiboo.se
Authentication-Results: phobos.denx.de; spf=pass
 smtp.mailfrom=bounces+31435339-7456-u-boot=lists.denx.de@em2124.kwiboo.se
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=kwiboo.se;
 h=from:subject:in-reply-to:references:mime-version:to:cc:
 content-transfer-encoding:content-type:cc:content-type:from:subject:to;
 s=s1; bh=YFjUB3DU4XZQxDBui1yLgQHTkWrZXppLudNr5CtAjs4=;
 b=ipoZi0ujGeWGbz4cV2Q1qPeeXuoNaGH4fF4TiWu3i+XvKHyyN6glc0GVEZ3Bqh9GT1Ro
 flIzYqCkcdCHyVdkz54xTiftInmMsxF0d+mOj1uzQgRKFGfDT4WtHI0zm+uR+vzsnf7bDH
 ratX39T+zWkMMZa9yFKPZaR/8KQYb0k6J/3BZH26gsPsY0kI24jKBV6/xKGTa8dksATu5H
 ErLn9TgB38pjv8EOlmJ7JBU507HGLDzkhCo2JyM0gsbLZXgTaCn90xlaIcj3fZxaxIxp2y
 TEyRy9jTdpxqpRfaBvqvryVKcBcULIJikSk4mxxZgsi4DuVu7FAIfZk8OW2Ld8Lw==
Received: by filterdrecv-59cb65cf6d-qj8b6 with SMTP id
 filterdrecv-59cb65cf6d-qj8b6-1-640FC204-15
 2023-03-14 00:38:28.669918367 +0000 UTC m=+458846.715192769
Received: from bionic.localdomain (unknown) by geopod-ismtpd-8 (SG) with ESMTP
 id sUPfWlXnQnKeNUnlnHn1Cw Tue, 14 Mar 2023 00:38:28.415 +0000 (UTC)
From: Jonas Karlman <jonas@kwiboo.se>
Subject: [PATCH 06/12] rockchip: rk3588-rock-5b: Fix sdmmc boot
Date: Tue, 14 Mar 2023 00:38:28 +0000 (UTC)
Message-Id: <20230314003755.512696-7-jonas@kwiboo.se>
X-Mailer: git-send-email 2.39.2
In-Reply-To: <20230314003755.512696-1-jonas@kwiboo.se>
References: <20230314003755.512696-1-jonas@kwiboo.se>
MIME-Version: 1.0
X-SG-EID: 
 TdbjyGynYnRZWhH+7lKUQJL+ZxmxpowvO2O9SQF5CwCVrYgcwUXgU5DKUU3QxAfZekEeQsTe+RrMu3cja6a0h9iwD0pPrCUEggNb7hbqr2zQkyTPykH/0+gmnEyvq7n93DJpXgQOkAWV0fmL7G8VrOnQg+08nrU+Z+hiukSRhvcAGaiv0x8WhjqAAb+rNf/UxD3NN5/mqd15teI6Wi2zEpNp1ikd0dQwQxOsdTvtemfSLQLpLVezRtT0B0gq3dGl
To: Kever Yang <kever.yang@rock-chips.com>, Simon Glass <sjg@chromium.org>,
 Philipp Tomsich <philipp.tomsich@vrull.eu>, Eugen Hristev
 <eugen.hristev@collabora.com>
Cc: Jagan Teki <jagan@edgeble.ai>, u-boot@lists.denx.de, Jonas Karlman
 <jonas@kwiboo.se>
X-Entity-ID: P7KYpSJvGCELWjBME/J5tg==
X-BeenThere: u-boot@lists.denx.de
X-Mailman-Version: 2.1.39
Precedence: list
List-Id: U-Boot discussion <u-boot.lists.denx.de>
List-Unsubscribe: <https://lists.denx.de/options/u-boot>,
 <mailto:u-boot-request@lists.denx.de?subject=unsubscribe>
List-Archive: <https://lists.denx.de/pipermail/u-boot/>
List-Post: <mailto:u-boot@lists.denx.de>
List-Help: <mailto:u-boot-request@lists.denx.de?subject=help>
List-Subscribe: <https://lists.denx.de/listinfo/u-boot>,
 <mailto:u-boot-request@lists.denx.de?subject=subscribe>
Errors-To: u-boot-bounces@lists.denx.de
Sender: "U-Boot" <u-boot-bounces@lists.denx.de>
X-Virus-Scanned: clamav-milter 0.103.8 at phobos.denx.de
X-Virus-Status: Clean

Running U-Boot from a SD-card on ROCK 5 Model B fails to load atf using
DMA and prints debug_uart messages.

  <debug_uart>

  <debug_uart>

  U-Boot SPL 2023.04-rc3 (Mar 12 2023 - 00:30:16 +0000)
  Trying to boot from MMC1
  ## Checking hash(es) for config config-1 ... OK
  ## Checking hash(es) for Image atf-1 ... sha256 error!
  Bad hash value for 'hash' hash node in 'atf-1' image node
  mmc_load_image_raw_sector: mmc block read error
  SPL: failed to boot from all boot devices
  ### ERROR ### Please RESET the board ###

Use fifo-mode to disable DMA in SPL, add same-as-spl to boot-order and
remove DEBUG_UART_ANNOUNCE option to fix this.

Signed-off-by: Jonas Karlman <jonas@kwiboo.se>
Reviewed-by: Kever Yang <kever.yang@rock-chips.com>
---
 arch/arm/dts/rk3588-rock-5b-u-boot.dtsi | 3 ++-
 configs/rock5b-rk3588_defconfig         | 1 -
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/dts/rk3588-rock-5b-u-boot.dtsi b/arch/arm/dts/rk3588-rock-5b-u-boot.dtsi
index 2386edf90deb..36d557b4934d 100644
--- a/arch/arm/dts/rk3588-rock-5b-u-boot.dtsi
+++ b/arch/arm/dts/rk3588-rock-5b-u-boot.dtsi
@@ -11,12 +11,13 @@
 	};
 
 	chosen {
-		u-boot,spl-boot-order = &sdmmc;
+		u-boot,spl-boot-order = "same-as-spl", &sdmmc;
 	};
 };
 
 &sdmmc {
 	bus-width = <4>;
 	u-boot,dm-spl;
+	u-boot,spl-fifo-mode;
 	status = "okay";
 };
diff --git a/configs/rock5b-rk3588_defconfig b/configs/rock5b-rk3588_defconfig
index 66199387195a..3fcc6a26bb51 100644
--- a/configs/rock5b-rk3588_defconfig
+++ b/configs/rock5b-rk3588_defconfig
@@ -66,6 +66,5 @@ CONFIG_PWM_ROCKCHIP=y
 CONFIG_SPL_RAM=y
 CONFIG_BAUDRATE=1500000
 CONFIG_DEBUG_UART_SHIFT=2
-CONFIG_DEBUG_UART_ANNOUNCE=y
 CONFIG_SYSRESET=y
 CONFIG_ERRNO_STR=y
